### DISCUSSION TITLE
make error on class libcamera::CameraConfiguration

### DISCUSSION THREAD

**Discussion by EloMatze79** [Motion-Project/motionplus#117] [Q&A]
Hello, 
i have tried to build the motionplus on a PI4 with the latest Raspbian OS and libcam.
The build works without problems on an old PI3 with no libcam support.

But on the newer PI i got following error:

libcam.cpp: In member function ‘void cls_libcam::cam_config_transform()’:
libcam.cpp:503:13: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  503 |     config->transform = Transform::Identity;
      |             ^~~~~~~~~
libcam.cpp:509:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  509 |                 config->transform = Transform::Identity;
      |                         ^~~~~~~~~
libcam.cpp:511:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  511 |                 config->transform = Transform::Rot0;
      |                         ^~~~~~~~~
libcam.cpp:513:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  513 |                 config->transform = Transform::HFlip;
      |                         ^~~~~~~~~
libcam.cpp:515:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  515 |                 config->transform = Transform::VFlip;
      |                         ^~~~~~~~~
libcam.cpp:517:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  517 |                 config->transform = Transform::HVFlip;
      |                         ^~~~~~~~~
libcam.cpp:519:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  519 |                 config->transform = Transform::Rot180;
      |                         ^~~~~~~~~
libcam.cpp:521:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  521 |                 config->transform = Transform::Transpose;
      |                         ^~~~~~~~~
libcam.cpp:523:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  523 |                 config->transform = Transform::Rot270;
      |                         ^~~~~~~~~
libcam.cpp:525:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  525 |                 config->transform = Transform::Rot90;
      |                         ^~~~~~~~~
libcam.cpp:527:25: error: ‘class libcamera::CameraConfiguration’ has no member named ‘transform’
  527 |                 config->transform = Transform::Rot180Transpose;
      |                         ^~~~~~~~~
make[2]: *** [Makefile:498: libcam.o] Error 1

Can anyone relate this issue?
Any solutions for this?

Thanks
Matthias

**Comment by EloMatze79** [comment]
I solved it.
libcamera0 was not installed correctly

**Reply by handbrianc** [reply]
I too am hitting the same issue.  what exactly did you do to solve your issue.  I did a apt install libcamera0-dev and nojoy.

**Comment by EloMatze79** [comment]
Hi, i installed via apt-get install the libcamera0.
Raspbian didn´t find libcamera0-dev on my device.

The problem is the code compiled with the libcamera0 install, but the motionplus didn´t start anyway.

Now i have this error:

Dec 23 09:37:38 [NTC][ALL][00:motionplus] log_init: MotionPlus 0.1.1-git-20231124-2cfd106 started
Dec 23 09:37:38 [NTC][ALL][00:motionplus] log_init: Using log type (ALL) log level (NTC)
Dec 23 09:37:38 [NTC][ALL][00:motionplus] mytranslate_init: Language: English
Dec 23 09:37:38 [NTC][ALL][00:motionplus] motpls_daemon: MotionPlus going to daemon mode
Dec 23 09:37:38 [NTC][ALL][00:motionplus] motpls_startup: MotionPlus running as daemon process
Dec 23 09:37:38 [NTC][STR][00:motionplus] webu_init_webcontrol: Starting webcontrol on port 8080
Dec 23 09:37:38 [NTC][STR][00:motionplus] webu_init_webcontrol: Started webcontrol on port 8080
Dec 23 09:37:38 [NTC][ENC][00:motionplus] motpls_av_init: libavcodec  version 59.37.100
Dec 23 09:37:38 [NTC][ENC][00:motionplus] motpls_av_init: libavformat version 59.27.100
Dec 23 09:37:38 [NTC][ALL][00:motionplus] main: Motionplus pid: 2336
Dec 23 09:37:38 [NTC][VID][00:motionplus] libcam_start: libcam not available
Dec 23 09:37:38 [ERR][ALL][00:motionplus] mlp_init_cam_start: Failed to start camera.
Dec 23 09:37:38 [NTC][ALL][00:motionplus] mlp_ring_resize: Resizing buffer to 1 items
Dec 23 09:37:38 [ERR][ALL][00:motionplus] mlp_init_firstimage: Unable to open camera


I configured the cam like this:  

libcam_device /dev/video0


I don´t no the problem at the moment...
